### PR TITLE
style(app,blueprints): add some type hints

### PR DESCRIPTION
**Pull request by SaidBySolo**
I'm using pylance's strict mode.

This PR makes my linter a little less complaining.

### CODE REVIEW

File: sanic/app.py [hunk 1]
```diff
@@ -411,7 +411,11 @@ def enable_websocket(self, enable=True):
 
         self.websocket_enabled = enable
 
-    def blueprint(self, blueprint, **options):
+    def blueprint(
+        self,
+        blueprint: Union[Blueprint, list[Blueprint], tuple[Blueprint], BlueprintGroup],
```
Comment (ahopkins) [inline sanic/app.py]: We should use `List` and `Tuple` since we also need to support 3.7

File: sanic/app.py [hunk 2]
```diff
@@ -869,7 +873,7 @@ def run(
         *,
         debug: bool = False,
         auto_reload: Optional[bool] = None,
-        ssl: Union[dict, SSLContext, None] = None,
+        ssl: Union[dict[str,str], SSLContext, None] = None,
```
Comment (ahopkins) [inline sanic/app.py:877]:
```suggestion
        ssl: Union[Dict[str, str], SSLContext, None] = None,
```

File: sanic/app.py [hunk 3]
```diff
@@ -999,7 +1003,7 @@ async def create_server(
         port: Optional[int] = None,
         *,
         debug: bool = False,
-        ssl: Union[dict, SSLContext, None] = None,
+        ssl: Union[dict[str,str], SSLContext, None] = None,
```
Comment (ahopkins) [inline sanic/app.py:1007]: Same

File: sanic/blueprints.py [hunk 4]
```diff
@@ -146,10 +146,10 @@ def signal(self, event: str, *args, **kwargs):
 
     @staticmethod
     def group(
-        *blueprints,
-        url_prefix="",
-        version=None,
-        strict_slashes=None,
+        *blueprints: Union[Blueprint, BlueprintGroup],
+        url_prefix: Optional[str] = "",
```
Comment (ahopkins) [inline sanic/blueprints.py:150]: I am not sure we do accept `None` as a value. I think this should only be `str`.

**Review by ahopkins — CHANGES_REQUESTED** [review]

File: sanic/app.py [hunk 5]
```diff
@@ -411,7 +412,11 @@ def enable_websocket(self, enable=True):
 
         self.websocket_enabled = enable
 
-    def blueprint(self, blueprint, **options):
+    def blueprint(
+        self,
+        blueprint: Union[Blueprint, List[Blueprint], Tuple[Blueprint], BlueprintGroup],
```
Comment (ahopkins) [inline sanic/app.py:417]: Instead of `List` and `Tuple`, maybe just `Sequence`

File: sanic/blueprints.py (hunk 4 above)
Comment (SaidBySolo) [reply sanic/blueprints.py:150]: Referenced here. Which one would be more accurate?

https://github.com/sanic-org/sanic/blob/ac784759d56428bcb714eab59caeb92f58391b96/sanic/blueprints.py#L72

Comment (ahopkins) [reply sanic/blueprints.py:150]: Hmm... maybe an inconsistency? Let me jump into and see if there is a reason it is `""` for a default there and not `None`.

Comment (ahopkins) [reply sanic/blueprints.py:150]: Looks like an inconsistency. The signature in `BlueprintGroup` is correct. Looking at the implementation in `Sanic.blueprint`, a `None` value is safe.

So, let's keep it as `Optional`, but also update the default value to `None`.

**Review by ahopkins — APPROVED** [review]
Thanks for putting this together. :clap:

File: sanic/app.py (hunk 5 above)
Comment (SaidBySolo) [inline sanic/app.py:417]: Maybe i have to specify it separately as a variable?

Comment (ahopkins) [reply sanic/app.py:417]: Run `make pretty`

Comment (SaidBySolo) [reply sanic/app.py]: Done ! :D

**Review by ahopkins — APPROVED** [review]

**Pull request merged**